fix: change product stream product export delete rule (#15243)
shopware/platform: RELEASE_INFO-6.7.md

## Core

### Product stream deletion is blocked while product exports exist

Deleting a product stream that's been used in a product export raises a dedicated delete restriction.

> In 6.7, this rule is additionally enforced on database level by changing the foreign key delete action from `CASCADE` to `RESTRICT`.

Reviewer: We don't need to mention 6.7 here 

```suggestion
This rule is also enforced on the database level by changing the foreign key delete action from `CASCADE` to `RESTRICT`.
```

## Core

### Product stream deletion is blocked while product exports exist

> Deleting a product stream now raises a dedicated delete restriction while related `product_export` records still exist.

Reviewer: ```suggestion
Deleting a product stream that's been used in a product export raises a dedicated delete restriction.
```

> This prevents accidental product export removals and makes the dependency visible in API and administration workflows.

Reviewer: This line can be removed as I don't think it bring any benefit

> In 6.8, this rule is additionally enforced on database level by changing the foreign key delete action from `CASCADE` to `RESTRICT`.

Reviewer: Is the last line really true? the migration is in 6_7
